docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -29,30 +30,93 @@ Include it in your module file like this:
+  * ``required:``
+    Only needed if true, otherwise it is assumed to be false.
+  * ``default:``
+    * If `required` is false/missing, `default` may be specified (assumed 'null' if missing).
+    * Ensure that the default parameter in docs matches default parameter in code. The default option must not be listed as part of the description.
+  * ```choices``
+    List of option values. Should be absent if empty.
+  * ``aliases:``
+    List of option name aliases, generally not needed. Should be absent if empty.
+  * ``version_added:``
+    Only needed if this option was extended after initial Ansible release, i.e. this is greater than the top level `version_added` field.
+    This is a string, and not a float, i.e. `version_added: "2.3"`
+* ``requirements:``
+    List of requirements, and minimum versions (if applicable). These are the requirements on the host that executes the module.
+* ``notes:``
+    Details of any important information that doesn't fit in one of the above sections, for example if ``check_mode`` isn't supported, or a link to external documentation.
+
+.. note:: Versions should be strings
+
+    In general, care should be taken to list versions as strings rather than floats.  This has
+    several reasons:
+    * Versions may not always be floats, for instance "2.1.0" or "2.0alpha1".
+    * Versions may not compare correctly if used as a float.  A float will tempt
+       naive comparison between 2.3 and 2.12 which won't generate the correct
+       ordering.  Strings "2.3" and "2.12" require a bit of code which will let the
+       programmer do it correctly.
+
+
+EXAMPLES block
+''''''''''''''
+
+The EXAMPLES section, just like the documentation section, it is required for all new modules.
+
+Examples should demonstrate real world usage, and be written in multi-line YAML format in plain text in.
+
+Ensure that examples are kept in sync with the options during the PR review and any following code refactor.
+
+As per playbook best practice a `name:` should be specified.
 
-Examples should be written in YAML format in plain text in an
 ``EXAMPLES`` string within the module like this::
 
     EXAMPLES = '''
-    - modulename:
-        opt1: arg1
-        opt2: arg2
+    - name: Ensure foo is installed
+      modulename:
+        name: foo
+        state: present
     '''
 
-The EXAMPLES section, just like the documentation section, is required in
-all module pull requests for new modules.
+If the module returns facts that are often needed an example of how to use them can be helpful.
 
-The RETURN section documents what the module returns. For each value returned,
-provide a ``description``, in what circumstances the value is ``returned``,
-the ``type`` of the value and a ``sample``.  For example, from
-the ``copy`` module::
+
+RETURN Block
+''''''''''''
+
+The RETURN section documents what the module returns, it is required for all new modules.
```

Comment:
The RETURN section documents what the module returns, and is required for all new modules.